biomage/stage/stage.py

```diff
@@ -202,13 +202,14 @@ def get_sandbox_id(templates, manifests, all_experiments, auto):
             if opts.ref != DEFAULT_BRANCH
         ]
     )
+    user_name = re.sub(r"[^\w\s]", "", os.getenv("BIOMAGE_NICK", os.getenv("USER", "")))
 
     # if we are in auto mode (non interactive) just generate a random sandbox ID name
     if auto:
-        return random_name.generate_name()
+        return (user_name + "-" + random_name.generate_name()).lower()[:26]
```

Comment:
I think you can't cut with `[:26]` because if it ends in a `-` then it's gonna be an invalid name iirc

Reply:
I recall something like that, but it does not make a lot of sense. Maybe just give it a try

Reply:
You're right, Kubernetes fails to create the namespace : 

```
2022-03-25 09:55:18 [ℹ] eksctl version 0.88.0 2022-03-25 09:55:18 [ℹ] using region eu-west-1 2022-03-25 09:55:21 [ℹ] creating Fargate profile "worker-agi-" on EKS cluster "biomage-staging" Error: could not create fargate profiles: failed to create Fargate profile "worker-agi-" on EKS cluster "biomage-staging": failed to create Fargate profile "worker-agi-": InvalidParameterException: The namespace provided is invalid. { RespMetadata: { StatusCode: 400, RequestID: "29dcfa60-03df-4150-a18f-b9fdf9944b88" }, Message_: "The namespace provided is invalid." }
```

Thanks for the heads up Pol! I'll add a check to ensure that the last element isn't a dash

Reply:
As explained in detail : https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names

TLDR Kubernetes object names must : 

- contain no more than 253 characters
- contain only lowercase alphanumeric characters, '-' or '.'
- start with an alphanumeric character
- end with an alphanumeric character